// ngx-showdown/esm2015/showdown.module.js
import { NgModule } from '@angular/core';
import { ShowdownConfig } from './showdown-config.provider';
import { ShowdownConverter } from './showdown-converter.provider';
import { ShowdownComponent } from './showdown.component';
import { ShowdownPipe } from './showdown.pipe';
import { SourceDirective } from './source.directive';
/**
 * @internal
 */
import * as ɵngcc0 from '@angular/core';
const DECLARATIONS = [
    ShowdownComponent,
    ShowdownPipe,
    SourceDirective
];
/**
 * ### Example
 *
 * Add `ShowdownModule` to app `imports`.
 * ```typescript
 * import { NgModule } from '@angular/core';
 * import { ShowdownModule } from 'ngx-showdown';
 *
 * @NgModule({
 *   imports: [ ShowdownModule ];
 * })
 * export class AppModule {}
 * ```
 */
export class ShowdownModule {
    /**
     * __Example :__
     *
     * Add `ShowdownModule` to app `imports` with config.
     * ```typescript
     * import { NgModule } from '@angular/core';
     * import { ShowdownModule } from 'ngx-showdown';
     *
     * @NgModule({
     *   imports: [ ShowdownModule.forRoot({
     *     smartIndentationFix: true,
     *     foo: 'bar',
     *     flavor: 'github',
     *     extensions: [ {type:'listener', listeners: {'codeBlocks.after': console.log}} ]
     *   }) ];
     * })
     * export class AppModule {}
     * ```
     * @param config - A root converter config for all converter instances.
     */
    static forRoot(config) {
        return {
            ngModule: ShowdownModule,
            providers: [{ provide: ShowdownConfig, useValue: config }]
        };
    }
}
ShowdownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ShowdownModule });
ShowdownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ShowdownModule_Factory(t) { return new (t || ShowdownModule)(); }, providers: [ShowdownConverter] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ShowdownModule, { declarations: [ShowdownComponent,
        ShowdownPipe,
        SourceDirective], exports: [ShowdownComponent,
        ShowdownPipe,
        SourceDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ShowdownModule, [{
        type: NgModule,
        args: [{
                declarations: DECLARATIONS,
                providers: [ShowdownConverter],
                exports: DECLARATIONS
            }]
    }], null, null); })();
