// ngx-showdown/esm2015/showdown-converter.provider.js
import { Injectable, Optional } from '@angular/core';
import * as Showdown from 'showdown';
import { ShowdownConfig } from './showdown-config.provider';
/**
 * @internal
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './showdown-config.provider';
let { hasOwnProperty } = {};
/**
 * ### Example
 *
 * Setup as standalone
 * ```typescript
 * import { NgModule } from '@angular/core';
 * import { ShowdownConverter } from 'ngx-showdown';
 *
 * @NgModule({
 *   providers: [ ShowdownConverter ];
 * })
 * export class AppModule {}
 * ```
 *
 * Use the converter instance.
 * ```typescript
 * import { Injectable } from '@angular/core';
 * import { ShowdownConverter } from 'ngx-showdown';
 *
 * @Injectable()
 * export class SomeService {
 *   constructor(showdownConverter: ShowdownConverter) {
 *     let markdown: string = "**Some**";
 *     let html: string = showdownConverter.makeHtml(markdown);
 *     console.log(`some:\nmarkdown: ${markdown)\nhtml: ${html}\n`);
 *   }
 * }
 * ```
 */
export class ShowdownConverter extends Showdown.Converter {
    constructor(config) {
        super(config && { extensions: config.extensions });
        this.setFlavor((config && config.flavor) || 'vanilla');
        this.setOptions(config);
    }
    /**
     * Set options to the converter.
     *
     * @param options - A options object to set.
     */
    setOptions(options) {
        for (let key in options) {
            if (hasOwnProperty.call(options, key)) {
                this.setOption(key, options[key]);
            }
        }
    }
}
ShowdownConverter.ɵfac = function ShowdownConverter_Factory(t) { return new (t || ShowdownConverter)(ɵngcc0.ɵɵinject(ɵngcc1.ShowdownConfig, 8)); };
ShowdownConverter.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: ShowdownConverter, factory: ShowdownConverter.ɵfac });
ShowdownConverter.ctorParameters = () => [
    { type: ShowdownConfig, decorators: [{ type: Optional }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ShowdownConverter, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.ShowdownConfig, decorators: [{
                type: Optional
            }] }]; }, null); })();
