// ngx-showdown/esm2015/source.directive.js
import { HttpClient } from '@angular/common/http';
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { ShowdownComponent } from './showdown.component';
/**
 * A angular directive to `ShowdownComponent` for make http request of markdown content.
 *
 * ### Example
 *
 * Setup as standalone
 * ```typescript
 * import { NgModule } from '@angular/core';
 * import { HttpClientModule } from '@angular/common/http';
 * import { ShowdownComponent, SourceDirective } from 'ngx-showdown';
 *
 * @NgModule({
 *    declarations: [ ShowdownComponent, SourceDirective ],
 *    imports: [ HttpClientModule ]
 * })
 * export class AppModule {}
 * ```
 *
 * Bind url `src` directive
 * ```typescript
 * import { Component } from '@angular/core';
 *
 * @Component({
 *     selector: 'some',
 *     template: '<showdown [src]="url" smartIndentationFix>**Loading...**</showdown>
 * })
 * class SomeComponent {
 *     url: string = 'https://unpkg.com/ngx-showdown/README.md';
 *     // ...
 * }
 * ```
 *
 * Set static url
 * ```html
 * <showdown src="README.md" [options]="{noHeaderId: true}"></showdown>
 * ```
 *
 * Set template reference variable
 * ```html
 * <showdown #source="source" src="README.md"></showdown>
 * ```
 *
 * Listening to `error` events.
 * ```html
 * <showdown #sd src="http://url.error" (error)="sd.render('# '+$event.message)"></showdown>
 * ```
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './showdown.component';
import * as ɵngcc2 from '@angular/common/http';
export class SourceDirective {
    constructor(_showdownComponent, _http) {
        this._showdownComponent = _showdownComponent;
        this._http = _http;
        /**
         * On error occur.
         *
         * __Example :__
         *
         * ```html
         * <input type="text" placeholder="url" [(ngModel)]="url"/>
         * <showdown [src]="url" (error)="sd.render('# Error\n> '+$event.message)">**Loading...**</showdown>
         * ```
         */
        this.error = new EventEmitter();
    }
    /**
     * A angular lifecycle method, Use to call to `load` method on src init/changes
     * @internal
     */
    ngOnChanges() {
        this.load();
    }
    /**
     * Load the markdown content of {@link SourceDirective#src} url to {@link ShowdownComponent#value}.
     *
     * __Example :__
     *
     * ```html
     * <input type="text" #url value="source.src" placeholder="Url" />
     * <button (click)="source.load(url.value)">Load</button>
     * <showdown #source="source" src="https://unpkg.com/ngx-showdown/README.md"></showdown>
     * ```
     * @param url - A url of markdown content to load (it will override the current url of `SourceDirective#src`)
     */
    load(url) {
        if (url) {
            this.src = url;
        }
        if (this.src) {
            this
                ._http
                .get(this.src, { responseType: 'text' })
                .subscribe((response) => {
                this._showdownComponent.render(response);
            }, (error) => {
                this.error.emit(error);
            });
        }
    }
}
SourceDirective.ɵfac = function SourceDirective_Factory(t) { return new (t || SourceDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ShowdownComponent), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.HttpClient)); };
SourceDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SourceDirective, selectors: [["showdown", "src", ""], ["", "showdown", "", "src", ""]], inputs: { src: "src" }, outputs: { error: "error" }, exportAs: ["source"], features: [ɵngcc0.ɵɵNgOnChangesFeature] });
SourceDirective.ctorParameters = () => [
    { type: ShowdownComponent },
    { type: HttpClient }
];
SourceDirective.propDecorators = {
    src: [{ type: Input }],
    error: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SourceDirective, [{
        type: Directive,
        args: [{
                selector: 'showdown[src],[showdown][src]',
                exportAs: 'source'
            }]
    }], function () { return [{ type: ɵngcc1.ShowdownComponent }, { type: ɵngcc2.HttpClient }]; }, { error: [{
            type: Output
        }], src: [{
            type: Input
        }] }); })();
